src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useContext, useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { TodosContext } from '../../TodosContext/TodosContext';
+import { FilteringBy } from '../../types/FilteringBy';
+
+export const Footer: React.FC = () => {
+  const { todos, handleClearCompleted, filteringBy, setFilteringBy } =
+    useContext(TodosContext);
+
+  const [countOfComplitedTodos, setCountOfComplitedTodos] = useState(0);
+  const [isAnyCompletedTodo, setIsAnyCompletedTodo] = useState(false);
+
+  useEffect(() => {
+    const complitedTodos = todos.filter(todo => !todo.completed).length;
+
+    setCountOfComplitedTodos(complitedTodos);
+
+    const isAnyCompleted = todos.some(todo => todo.completed);
+
+    setIsAnyCompletedTodo(isAnyCompleted);
+  }, [todos]);
```

Comment:
Don't store these values in the state, it doesn't make sense. Just store them in variables